app/Http/Controllers/CircleController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests;
+use App\Models;
+use Input;
+
+class CircleController extends Controller
+{
+    public function getCircles() {
+        return response()->success(Models\Circle::get());
+    }
+
+    public function getCircle($circle_id) {
+        return response()->success(Models\Circle::where('id', $circle_id)->with(['body', 'childrenCircles', 'parentCircle'])->first());
+    }
+
+    public function getCirclesOfBody($body_id) {
+        return response()->success(Models\Body::findOrFail($body_id)->circles);
+    }
+
+    public function getCirclesOfUser($user_id) {
+        return response()->success(Models\User::findOrFail($user_id)->circles);
+    }
+
+    //RECURSIVE METHODS
+    public function getCircleMembers($circle_id) {
```

Comment:
Do you think it would be wise to also provide a "flat" implementation of this method, as if you only want to know which members are in the circle directly?